modules/react/button/lib/ExternalHyperlink.tsx

```diff
@@ -41,18 +42,29 @@ const StyledSystemIcon = styled(SystemIcon)<StyledType>({
 export const ExternalHyperlink = createComponent('a')({
   displayName: 'ExternalHyperlink',
   Component: (
-    {children, iconLabel = 'Opens link in new window', ...elemProps}: ExternalHyperlinkProps,
-    ref
+    {
+      children,
+      iconLabel = 'Opens link in new window',
+      variant,
+      ...elemProps
+    }: ExternalHyperlinkProps,
+    ref,
+    Element
   ) => (
-    <Anchor ref={ref} target="_blank" rel="noreferrer" {...elemProps}>
-      <span>{children}</span>
-      <StyledSystemIcon
+    <Element
+      ref={ref}
+      target="_blank"
+      rel="noreferrer"
+      {...handleCsProp(elemProps, externalHyperlinkStencil({variant}))}
+    >
+      <span data-part="external-hyperlink-children">{children}</span>
+      <SystemIcon
         icon={extLinkIcon}
         role="img"
         aria-label={iconLabel}
-        size={iconSize}
-        cs={{'& svg': {minWidth: minIconSize, minHeight: minIconSize}}}
+        data-part="external-hyperlink-icon"
+        size={relativeToFontSizeIconSize}
```

Comment:
Should we move this to the base styles instead?

```js
    '& [data-part="external-hyperlink-icon"]': {
      [systemIconStencil.vars.color]: 'currentColor',
      [systemIconStencil.vars.size]: relativeToFontSizeIconSize // or why not '1em'?
```

Reply:
i like